### PR TITLE
Beditems track

### PR DESCRIPTION
**Pull request by pkerpedjiev**
## Description

What was changed in this pull request?

This PR adds lets you create a BedLike track from data that is already in a Python object using a track's `local-tiles` data.

Why is it necessary?

Fixes #___

## Checklist

- [x] Unit tests added or updated
- [x] Update `examples.ipynb` notebook
- [x] Documentation added or updated
- [x] Updated CHANGELOG.md
- [x] Ran `black` on the root directory

### CODE REVIEW

File: docs/getting_started.rst [hunk 1]
```diff
@@ -446,13 +446,13 @@ according to the chromosome info in the specified file.
     bigwig_fp = '../data/wgEncodeCaltechRnaSeqHuvecR1x75dTh1014IlnaPlusSignalRep2.bigWig'
 
     with open(chromsizes_fp) as f:
-        chromsizes = []
+        chromsizes_arr = []
```
Comment (flekschas) [inline docs/getting_started.rst:449]: Is this related to the Bed track?

File: higlass/trackdata.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    # we want the cumlengths to start 0
+    cumlengths = [0] + cumlengths[:-1]
+
+    return dict([(name, {
+        'name': name,
+        'length': length,
+        'start': start
+    }) for name, length, start in zip(chromnames, chromlengths, cumlengths)])
+
+def bedtiles(lines: List[Tuple[str, int, int, str, str, str]],
```
Comment (flekschas) [inline higlass/trackdata.py:31]: What is the difference between `trackdata` and tilesets?

**Review by flekschas — COMMENTED** [review]
Look 👍 I just don't quite get the semantic difference between creating a tileset and trackdata. Could you elaborate on that? I am asking because if I was a new user I would probably be confused how the two concepts are related and why there isn't just a tileset for bed data.

File: docs/getting_started.rst (hunk 1 above)
Comment (pkerpedjiev) [reply docs/getting_started.rst:449]: No but it's an error in the documentation so I thought I'd straighten it out here.

File: higlass/trackdata.py (hunk 2 above)
Comment (pkerpedjiev) [reply higlass/trackdata.py:31]: Trackdata does not require a server. It uses [local tiles whose values are encoded in the viewconf](https://docs.higlass.io/view_config.html#raw-tile-values).

Comment (pkerpedjiev) [reply higlass/trackdata.py:31]: I added a note to the docs briefly explaining this.

Comment (flekschas) [reply higlass/trackdata.py:31]: This is all just semantics but I find the naming confusing. Why not call it _local_ or _inline_ data? Especially in the context of the Track object where it's also just referred to as `data=data`. I assume `data` is inspired by the viewconf prop with the same name but I wonder how should a new user know that the `data` property is meant for a very special use case only?

I would propose to rename `data` to `local_tiles`, `trackdata` to `inline_data` or something similar.

Anyway, just my 2 cents. Not critical

File: higlass/trackdata.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    chroms: Union[Path, str, list], **kwargs) -> Track:
+    """Generate a list of local tiles that can be used with a bedlike track.
```
Comment (nvictus) [inline higlass/trackdata.py:33]: This seems to be raw sphinx docstrings instead of numpydoc. Shouldn't we be consistent?

File: higlass/trackdata.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    
+    """
+    if isinstance(chroms, Path) or isinstance(chroms, str):
+        chrom_names_lengths = chromsize_pairs(chroms)
+    if isinstance(chroms, list):
+        chrom_names_lengths = chroms
+    else:
+        ValueError(f"Unknown chroms type: {type(chroms)}. Expecting str, Path or list")
+
+    chroms = load_chromsizes(chrom_names_lengths)
```
Comment (nvictus) [inline higlass/trackdata.py:51]: Why force the user to pass in a file?

File: higlass/trackdata.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        'length': length,
+        'start': start
+    }) for name, length, start in zip(chromnames, chromlengths, cumlengths)])
+
+def bedtiles(lines: List[Tuple[str, int, int, str, str, str]],
+    chroms: Union[Path, str, list], **kwargs) -> Track:
+    """Generate a list of local tiles that can be used with a bedlike track.
+    
+    Args:
+        lines: A list of bed style lines ([chrom, start, end, name, score, pos])
```
Comment (nvictus) [inline higlass/trackdata.py:36]: No support for bed9 (`thickStart`, `thickEnd`, but most importantly `itemRgb`)?

File: higlass/trackdata.py (hunk 2 above)
Comment (nvictus) [reply higlass/trackdata.py:31]: I agree with @flekschas's naming suggestions. Maybe even call both of them `inline_tiles`.

File: higlass/trackdata.py (hunk 4 above)
Comment (nvictus) [reply higlass/trackdata.py:51]: Oh, I see that it could be a list of pairs. I don't see why it can't just be a dictionary (pandas Series are also dict-like and dicts are now always ordered as of py3.6).

File: higlass/trackdata.py (hunk 3 above)
Comment (pkerpedjiev) [reply higlass/trackdata.py:33]: Good catch. We should indeed be consistent. I'll update the docstring.

It's was meant to be [Google style docstrings](https://google.github.io/styleguide/pyguide.html#383-functions-and-methods).

File: higlass/trackdata.py (hunk 5 above)
Comment (pkerpedjiev) [reply higlass/trackdata.py:36]: There's support for itemRgb, it's just not necessary. `thickStart` and `thickEnd` are unfortunately not supported by higlass and thus not supported here.

I updated the docs to make a note of this.

File: higlass/trackdata.py (hunk 4 above)
Comment (pkerpedjiev) [reply higlass/trackdata.py:51]: It sure can. I added that as a supported option.

File: higlass/trackdata.py (hunk 2 above)
Comment (pkerpedjiev) [reply higlass/trackdata.py:31]: I'm on board with naming `trackdata` to `inline_data`. I'm a little hesitant to name the `Track` parameter `inline_tiles` because it can be used for other types of data (e.g. `divided` datasets or BAM file urls).
